projects/rero/ng-core/src/lib/record/detail/detail.component.ts

```diff
@@ -166,10 +176,26 @@ export class DetailComponent implements OnInit, OnDestroy {
   /**
    * Delete the record and go back to previous page.
    * @param event - DOM event
-   * @param pid - string, PID to remove
-   */
-  deleteRecord(pid: string) {
-    this._recordUiService.deleteRecord(this._route.snapshot.paramMap.get('type'), pid).subscribe((result: any) => {
+   * @param element - string (PID to remove) or object
+   */
+  deleteRecord(element: any) {
```

Comment:
This can certainly be rewritten by having only one `this._recordUiService.deleteRecord` in you code.

Reply:
done